### DISCUSSION TITLE
Vercel ignores error.tsx on status code 500 for Nextjs 13.4.19

### DISCUSSION THREAD

**Discussion by Metarock** [vercel/next.js#55130] [App Router]
## Environment Information 

```
   Operating System:
      Platform: linux
      Arch: x64
      Version: #202307241739~1692717645~22.04~5597803 SMP PREEMPT_DYNAMIC Tue A
    Binaries:
      Node: 18.17.0
      npm: 9.6.7
      Yarn: 1.22.19
      pnpm: 8.6.11
    Relevant Packages:
      next: 13.4.19
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
    Next.js Config:
      output: standalone

```

## Expected Behavior 

When running in localhost, the error code 500 is caught and `error.tsx` works

![image](https://github.com/vercel/next.js/assets/66818636/c528f4a2-bee0-46d8-9e4a-89b0bde59952)

## Actual behavior

However when running on production, `yarn build && yarn start` or deployed to vercel. We get the default 500 page instead

![image](https://github.com/vercel/next.js/assets/66818636/a2ed711f-923f-4502-bb13-7eb0f7c8a882)

## Error message 

```Error
    at getEvent (/home/user/Projects/company/company-shopsite/.next/server/app/events/[...slug]/page.js:4164:34)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async EventPage (/home/user/Projects/company/company-shopsite/.next/server/app/events/[...slug]/page.js:4168:19)
Error
    at getEvent (/home/user/Projects/company/company-shopsite/.next/server/app/events/[...slug]/page.js:4164:34)
    at async Module.generateMetadata (/home/user/Projects/company/company-shopsite/.next/server/app/events/[...slug]/page.js:4245:19)
[Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: '927537468'
}
Error
    at getEvent (/home/user/Projects/company/company-shopsite/.next/server/app/events/[...slug]/page.js:4164:34)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async EventPage (/home/user/Projects/company/company-shopsite/.next/server/app/events/[...slug]/page.js:4168:19)
```

![image](https://github.com/vercel/next.js/assets/66818636/aa35cd15-b460-4094-9a4e-e01d04d2c286)

## Code for catching the error
```async function getEvent(slug: string): Promise<EventData> {
  const res = await fetch(
    `${process.env.NEXT_PUBLIC_BASE_URL}/legacy/${process.env.NEXT_PUBLIC_COUNTRY_CODE}/shop/events/slug/events/${slug}`,
    { next: { revalidate: 15 } }
  );

  if (res.status >= 400 && res.status < 500) notFound();

  if (res.status >= 500) throw new Error("Sorry, an error has occurred while fetching your event.");

  return res.json();
}
```

If the api's status code is 500 or anything else I expect it to throw an error and use the `error.tsx`. However, on production it is display the default 500 error page. Am I missing something here?

**Comment by magnnus** [comment]
I encountered the same problem

**Reply by magnnus** [reply]
Next.js v14.1.4, and use App Router.

When i set `revalidate = 0` (close the Cache strategy), the `error.tsx` triggered successfully ( **Dynamic render** will triiger the `error.tsx`).

Maybe the 500 status purpose  is to tell the browser or CDNs to not cache the content. But I still want to render `error.tsx` insteadof the `500 | internal server error` page.

**Reply by magnnus** [reply]
https://nextjs.org/docs/messages/app-static-to-dynamic-error

**Reply by magnnus** [reply]
https://stackoverflow.com/questions/77051205/how-can-i-create-custom-error-page-for-500-internal-server-error-in-next-js-app

But when the page active the cache, `redirect` can't set the `cache-control: no-store`

**Reply by magnnus** [reply]
Now i delete `loading.js` & `<Suspense>`， and then use  **Page Router** `_error.js` to control custom 500 UI.
